src/formatDate.js

```diff
@@ -8,7 +8,42 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let dateArr = [];
+  let resultString = '';
+  let index = 0;
+
+  dateArr = date.split(fromFormat[3]);
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const fromIndex = fromFormat.indexOf(toFormat[i]);
+
+    index++;
+
+    if (toFormat[i] === 'YY' && fromFormat.includes('YYYY')) {
+      const number = dateArr[index - 1];
+
+      resultString += number.slice(-2);
+    } else if (toFormat[i] === 'YYYY' && fromFormat.includes('YY')) {
```

Comment:
This condition checks if we need to convert from `YYYY` to `YY`. However, it doesn't check if the current format in `fromFormat` is actually `YYYY`. This can lead to incorrect conversions if `fromFormat` contains both `YYYY` and `YY`. The same issue is present in the condition for converting from `YY` to `YYYY`.